src/core/crypto/crypto_platform.cpp

```diff
@@ -298,7 +298,11 @@ OT_TOOL_WEAK otError otPlatCryptoSha256Start(void *aContext, size_t aContextSize
     mbedtls_sha256_context *context = static_cast<mbedtls_sha256_context *>(aContext);
 
     VerifyOrExit(aContextSize >= sizeof(mbedtls_sha256_context), error = kErrorFailed);
+#if (MBEDTLS_VERSION_NUMBER >= 0x03000000)
+    VerifyOrExit((mbedtls_sha256_starts(context, 0) == 0), error = kErrorFailed);
```

Comment:
Cant we define `mbedtls_xyz` calls to `mbedtls_xyz_ret` (for MBEDTLS_VERSION_NUMBER < 0x03000000) avoiding #ifs? We can do it in `crypto_platform.cpp` or `crypto.h`.

Reply:
I think it is possible, however, I don't have a strong opinion which of the approach is more cleaner. If we have mbedTLS 3.0 as a default option we could use this [file](https://github.com/ARMmbed/mbedtls/blob/v3.0.0/include/mbedtls/compat-2.x.h). Do you have some thoughts @jwhui ?

Reply:
I believe there are only three instances that would benefit from this cleanup. I'm OK to accept this PR as is and cleanup when we switch the default to mbedTLS 3.0.